pkgs/auto-fix-vscode-server.nix

```diff
@@ -1,94 +1,152 @@
-{ lib, writeShellScript, coreutils, findutils, inotify-tools, patchelf, nodejs-16_x, buildFHSUserEnv
-, nodejsPackage ? nodejs-16_x
+{ lib, buildFHSUserEnv
+, writeShellScript, coreutils, findutils, inotify-tools, patchelf
+, stdenv, curl, icu, libunwind, libuuid, lttng-ust, openssl, zlib, krb5
 , enableFHS ? false
-, extraFHSPackages ? (pkgs: [ ])
+, nodejsPackage ? null
+, extraRuntimeDependencies ? [ ]
 , installPath ? "~/.vscode-server"
 }:
 
 let
-  nodejs = nodejsPackage;
+  inherit (lib) makeBinPath makeLibraryPath optionalString;
 
   # Based on: https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/applications/editors/vscode/generic.nix
-  nodejsFHS = buildFHSUserEnv {
-    name = "node";
+  runtimeDependencies = [
+    stdenv.cc.libc
+    stdenv.cc.cc
 
-    # additional libraries which are commonly needed for extensions
-    targetPkgs = pkgs: (builtins.attrValues {
-      inherit (pkgs)
-        # ld-linux-x86-64-linux.so.2 and others
-        glibc
-
-        # dotnet
-        curl
-        icu
-        libunwind
-        libuuid
-        lttng-ust
-        openssl
-        zlib
-
-        # mono
-        krb5
-      ;
-    }) ++ extraFHSPackages pkgs;
+    # dotnet
+    curl
+    icu
+    libunwind
+    libuuid
+    lttng-ust
+    openssl
+    zlib
 
-    runScript = "${nodejs}/bin/node";
+    # mono
+    krb5
+  ];
 
+  nodejs = nodejsPackage;
+  nodejsFHS = buildFHSUserEnv ({
+    name = "node";
+    targetPkgs = _: runtimeDependencies;
+    extraBuildCommands = ''
+      if [[ -d /usr/lib/wsl ]]; then
+        cp -rsHf /usr/lib/wsl usr/lib/wsl
+      fi
+    '';
+    runScript = "${nodejs}/bin/node";
     meta = {
       description = ''
         Wrapped variant of Node.js which launches in an FHS compatible envrionment,
-        which should allow for easy usage of extensions without nix-specific modifications.
+        which should allow for easy usage of extensions without Nix-specific modifications.
       '';
     };
-  };
+  });
 
-  nodejsWrapped = if enableFHS then nodejsFHS else nodejs;
+  patchELFScript = writeShellScript "patchelf-vscode-server.sh" ''
+    set -euo pipefail
+    PATH=${makeBinPath [ coreutils findutils patchelf ]}
+    INTERP=$(< ${stdenv.cc}/nix-support/dynamic-linker)
+    RPATH=${makeLibraryPath runtimeDependencies}
+    bin_dir=$1
 
-in writeShellScript "auto-fix-vscode-server.sh" ''
-  set -euo pipefail
-  PATH=${lib.makeBinPath [ coreutils findutils inotify-tools patchelf ]}
+    # NOTE: We don't log here because it won't show up in the output of the user service.
 
-  bins_dir=${installPath}/bin
-  node_interp=$(patchelf --print-interpreter ${nodejs}/bin/node)
-  node_rpath=$(patchelf --print-rpath ${nodejs}/bin/node)
+    # Check if the installation is already full patched.
+    if (( $(< "$bin_dir/.patched") )); then
```

Comment:
> /home/\<SNIP\>/.vscode-server/bin/441438abd1ac652551dbe4d408dfcec8a499b8bf/.patched: No such file or directory

Reply:
Probably a result of something else going wrong. Though just in case I have added that defense check of the dir existing and being a dir.